cns/restserver/restserver.go

```diff
@@ -379,6 +383,115 @@ func (service *HTTPRestService) deleteNetwork(w http.ResponseWriter, r *http.Req
 	log.Response(service.Name, resp, resp.ReturnCode, ReturnCodeToString(resp.ReturnCode), err)
 }
 
+// Handles CreateHnsNetwork requests.
+func (service *HTTPRestService) createHnsNetwork(w http.ResponseWriter, r *http.Request) {
+	log.Printf("[Azure CNS] createHnsNetwork")
+
+	var err error
+	returnCode := 0
+	returnMessage := ""
+
+	var req cns.CreateHnsNetworkRequest
+	err = service.Listener.Decode(w, r, &req)
+	log.Request(service.Name, &req, err)
+
+	if err != nil {
+		returnMessage = fmt.Sprintf("[Azure CNS] Error. Unable to decode input request.")
+		returnCode = InvalidParameter
+	} else {
+		switch r.Method {
+		case "POST":
+			if err := platform.CreateHnsNetwork(req); err == nil {
+				// Save the newly created HnsNetwork name. CNS deleteHnsNetwork API
+				// will only allow deleting these networks.
+				networkInfo := &networkInfo{
+					NetworkName: req.NetworkName,
+				}
+				service.lock.Lock()
+				service.state.Networks[req.NetworkName] = networkInfo
+				service.lock.Unlock()
+				returnMessage = fmt.Sprintf("[Azure CNS] Successfully created HNS network: %s", req.NetworkName)
+			} else {
+				returnMessage = fmt.Sprintf("[Azure CNS] CreateHnsNetwork failed with error %v", err.Error())
+				returnCode = UnexpectedError
+			}
+		default:
+			returnMessage = "[Azure CNS] Error. CreateHnsNetwork did not receive a POST."
+			returnCode = InvalidParameter
+		}
+	}
+
+	resp := &cns.Response{
+		ReturnCode: returnCode,
+		Message:    returnMessage,
+	}
+
+	err = service.Listener.Encode(w, &resp)
+
+	if returnCode == 0 {
+		service.lock.Lock()
+		service.saveState()
+		service.lock.Unlock()
+	}
+
+	log.Response(service.Name, resp, resp.ReturnCode, ReturnCodeToString(resp.ReturnCode), err)
+}
+
+// Handles deleteHnsNetwork requests.
+func (service *HTTPRestService) deleteHnsNetwork(w http.ResponseWriter, r *http.Request) {
+	log.Printf("[Azure CNS] deleteHnsNetwork")
+
+	var err error
+	var req cns.DeleteHnsNetworkRequest
+	returnCode := 0
+	returnMessage := ""
+
+	err = service.Listener.Decode(w, r, &req)
+	log.Request(service.Name, &req, err)
+
+	if err != nil {
+		returnMessage = fmt.Sprintf("[Azure CNS] Error. Unable to decode input request.")
+		returnCode = InvalidParameter
+	} else {
+		switch r.Method {
+		case "POST":
+			service.lock.Lock()
+			networkInfo, ok := service.state.Networks[req.NetworkName]
```

Comment:
I will encapsulate this in a function, so that people don't forget to acquire lock.

Reply:
Keeping it as is. Don't understand the comment. I don't think moving the networkName validity check warrants a new func if that's what you are asking for.